Sources/hostmgr/commands/generate/GenerateBuildkiteJobScript.swift

```diff
@@ -27,56 +27,10 @@ struct GenerateBuildkiteJobScript: ParsableCommand {
         ].joined(separator: "\n")
     }
 
-    // swiftlint:disable:next function_body_length
     func generateExports(
         from environment: [String: String] = ProcessInfo.processInfo.environment
     ) throws -> [String: String] {
-
-        // See https://buildkite.com/docs/pipelines/environment-variables for
-        // and up-to-date list of environment variables that Buildkite exports
-        let copyableExports = [
-            "BUILDKITE",
-            "BUILDKITE_AGENT_DEBUG",
-            "BUILDKITE_AGENT_DEBUG_HTTP",
-            "BUILDKITE_AGENT_ENDPOINT",
-            "BUILDKITE_AGENT_INCLUDE_RETRIED_JOBS",
-            "BUILDKITE_AGENT_META_DATA_QUEUE",
-            "BUILDKITE_AGENT_NAME",
-            "BUILDKITE_AGENT_NO_COLOR",
-            "BUILDKITE_AGENT_PROFILE",
-            "BUILDKITE_ARTIFACT_PATHS",
-            "BUILDKITE_ARTIFACT_UPLOAD_DESTINATION",
-            "BUILDKITE_BOOTSTRAP_PHASES",
-            "BUILDKITE_BRANCH",
-            "BUILDKITE_BUILD_NUMBER",
-            "BUILDKITE_BUILD_URL",
-            "BUILDKITE_CLEAN_CHECKOUT",
-            "BUILDKITE_COMMIT",
-            "BUILDKITE_GIT_CLONE_FLAGS",
-            "BUILDKITE_HOOKS_PATH",
-            "BUILDKITE_JOB_ID",
-            "BUILDKITE_LABEL",
-            "BUILDKITE_MESSAGE",
-            "BUILDKITE_NO_HTTP2",
-            "BUILDKITE_ORGANIZATION_SLUG",
-            "BUILDKITE_PIPELINE_PROVIDER",
-            "BUILDKITE_PIPELINE_SLUG",
-            "BUILDKITE_PULL_REQUEST",
-            "BUILDKITE_PULL_REQUEST_REPO",
-            "BUILDKITE_REFSPEC",
-            "BUILDKITE_REPO",
-            "BUILDKITE_TAG",
-
-            // These ones aren't printed as part of the default list – we're copying them so that `bootstrap` works
-            "BUILDKITE_AGENT_ACCESS_TOKEN",
-            "BUILDKITE_BUILD_ID",
-            "BUILDKITE_PLUGINS_PATH"
-
-        ].reduce([String: String]()) { dictionary, key in
-            var mutableDictionary = dictionary
-            mutableDictionary[key] = environment[key]
-            return mutableDictionary
-        }
+        let copyableExports = environment.filter { $0.key.starts(with: "BUILDKITE") }
```

Comment:
Buildkite has a number of environment variables you can use with all sorts of different behavior, and I don't think we want/have to use them all. Maintaining the list might be a bit of a pain but I think it is the correct approach IMO.

Reply:
We don't need to use them all, but as we discovered with #23, sometimes we want more and we don't know that they might cause problems. I've opened #29 which will hopefully give us a bit more confidence that we've handled both current and potential future cases. 

Initially I went for a more careful approach here, but we still ended up with breakage, so given the fact that everything Buildkite-related is namespaced as such, I'm not sure the benefit is still there.

Reply:
@rynaardb do you mind having another look at this PR, which has changed quite a bit since Jeremy has refactored the script generation code?